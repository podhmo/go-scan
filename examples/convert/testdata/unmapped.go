package testdata

// @derivingconvert(UnmappedDst)
type UnmappedSrc struct {
	ID   int
	Name string
}

type UnmappedDst struct {
	ID          int
	Name        string
	Extra       string  // This should be in the docstring
	Another     int     // This should also be in the docstring
	Optional    *string // This should be ignored
	ExtraPtr    *bool   // This should also be ignored
}
