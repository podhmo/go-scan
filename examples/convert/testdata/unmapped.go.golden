// Code generated by convert. DO NOT EDIT.
package testdata

import (
	"context"
	"errors"
	"fmt"
	"github.com/podhmo/go-scan/examples/convert/model"
)

// The following fields are not filled by this conversion: Extra, Another
func convertUnmappedSrcToUnmappedDst(ctx context.Context, ec *model.ErrorCollector, src *UnmappedSrc) *UnmappedDst {
	if src == nil {
		return nil
	}
	dst := &UnmappedDst{}
	if ec.MaxErrorsReached() {
		return dst
	}
	ec.Enter("ID")
	dst.ID = src.ID
	ec.Leave()
	if ec.MaxErrorsReached() {
		return dst
	}
	ec.Enter("Name")
	dst.Name = src.Name
	ec.Leave()
	return dst
}

// ConvertUnmappedSrcToUnmappedDst converts UnmappedSrc to UnmappedDst.
func ConvertUnmappedSrcToUnmappedDst(ctx context.Context, src *UnmappedSrc) (*UnmappedDst, error) {
	if src == nil {
		return nil, nil
	}
	ec := model.NewErrorCollector(0)
	dst := convertUnmappedSrcToUnmappedDst(ctx, ec, src)
	if ec.HasErrors() {
		return dst, errors.Join(ec.Errors()...)
	}
	return dst, nil
}
